test-suite/shared/src/test/scala/org/scalajs/testsuite/scalalib/RangesTest.scala

```diff
@@ -35,4 +37,16 @@ class RangesTest {
 
     assertEquals(List(0.0, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0), range.toList)
   }
+
+  @Test def NumericRange_overflow_issue_2407(): Unit = {
+    val nr = NumericRange(Int.MinValue, Int.MaxValue, 1 << 23)
+    assertEquals(Int.MinValue, nr.sum)
+  }
+
+  @Test def Range_foreach_issue_2409(): Unit = {
+    val r = Int.MinValue to Int.MaxValue by (1<<23)
+    assertEquals(512, r.length)
+    assertEquals(512, { var i = 0; r.foreach(_ => i += 1); i })
```

Comment:
I suggest writing this outside the `assertEquals`, with a somewhat normal style.